src/execution/sql/functions/string_functions.cpp

```diff
@@ -9,20 +9,27 @@
 
 namespace terrier::execution::sql {
 
-void StringFunctions::Concat(StringVal *result, exec::ExecutionContext *ctx, const StringVal &left,
-                             const StringVal &right) {
-  if (left.is_null_ || right.is_null_) {
+void StringFunctions::Concat(StringVal *result, exec::ExecutionContext *ctx, const int64_t num_inputs,
+                             const StringVal *inputs) {
```

Comment:
The `inputs` array needs to store *pointers* to string values to avoid copying each string value into the array. So:

```
Concat(StringVal *result, ExecutionContext *ctx, const StringVal *inputs[], uint32_t num_inputs)
```

Reply:
updated